### PR TITLE
Provide inputs for releaser and manifest options in release-please 13.15

### LINKED ISSUE #427: Feat: Support GitHub release notes generation
### TL;DR

v13 of the release-please project [added support](https://github.com/googleapis/release-please/pull/1120) for the new [GitHub API](https://docs.github.com/en/rest/reference/releases#generate-release-notes-content-for-a-release) that generates release notes: .  This would be great to have access to in this action.

### Detailed design

```markdown
The new release notes generation API is exposed in the release-please library as well as the CLI, however this action does not support it.  Their docs refer to it as Changelog Type but the code also uses ChangelogNotesType, with the existing implementation being called `default` and the new type being called `github`.

Complicating matters, this action already has an input named `changelogTypes` although it might have been better named `changelogSections` as that is what it gets passed to in the library, see https://github.com/google-github-actions/release-please-action/blob/main/index.js#L132.  

There's a parity with upstream vs. backwards compatibility decision here of whether to rename the existing `changelogTypes` input.  Assuming no, the next question is whether adding `changelogType` would be too confusing or possibly to use `changelogNotesType` instead.
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by andrewesweet**
Originally I set out to resolve #427. Along the way I noticed a number of other settings in both ReleaserConfig and ManifestOptions that were not exposed by this action or were inconsistently set between the two. On that basis, I've exposed all (I think) ReleaserConfig and ManifestOptions attributes as action inputs. I've taken the descriptions from the release-please CLI and manifest releaser docs wherever possible. One or two were poorly documented and came from code comments.

In the context of #427, 'changelog type' has already been given a different meaning by the action than in release-please, so I opted for 'changelog-notes-type' instead.

I think #451 and #473 are resolved or subsumed by this PR.

### CODE REVIEW

**Comment by andrewesweet** [comment]
I've noticed some of the input types don't line up. Will fix.

**Comment by andrewesweet** [comment]
@bcoe Is there anything I can do to help this PR through, please?

File: action.yml [hunk 1]
```diff
@@ -96,6 +96,78 @@ inputs:
     description: 'mark pull request as a draft'
     required: false
     default: false
+  changelog-notes-type:
+    description: "Strategy for building the changelog contents(see https://github.com/googleapis/release-please/blob/main/docs/customizing.md#changelog-types). Default `default`. Called `changelog-type` in release-please documentation."
+    required: false
+  changelog-host:
+    description: 'Host for commit hyperlinks in the changelog. Default `https://github.com`'
```
Comment (chingor13) [inline action.yml:103]: I think we need to add a `required: false` here

**Review by chingor13 — APPROVED** [review]
This looks reasonable to me

**Comment by bcoe** [comment]
@andrewesweet bother you to address @chingor13 one nit?

**Comment by andrewesweet** [comment]
> @andrewesweet bother you to address @chingor13 one nit?

Thanks for the feedback. I’m enjoying a vacation at the moment. I’ll pick this up when I return.

File: action.yml (hunk 1 above)
Comment (chingor13) [reply action.yml]: This was added separately in #550

**Review by chingor13 — APPROVED** [review]

**Pull request merged**